### ISSUE TITLE
Avoiding Overfitting

### ISSUE THREAD

**Issue by ikvision** [nkolot/SPIN#45]
During the training TensorBoard shows the training loss. 
Without a validation loss I find it difficult to estimate if the system overfits. Therefore It is difficult to know when training should stoped or further regularized. 

Seems like the BaseTrainer has a dedicated function for it:
https://github.com/nkolot/SPIN/blob/6de0944655721e8b2ecad6566aa40bc436a0e662/utils/base_trainer.py#L75:L77
Not implemented in its inheriting class BaseTrainer

I only see test set for final evaluation, am I missing the option to incorporate a validation set?
What method do you use to pick the best model?

**Comment by geopavlakos** [comment]
Considering the variety of images in the training data, we haven't really observed any overfitting behavior. For validation, the [3DPW dataset](https://virtualhumans.mpi-inf.mpg.de/3DPW/) offers a good option for a validation set.

**Comment by ikvision** [comment]
Thank you for the answer, I don't see overfitting either.
Did you pick the best model using the best performing checkpoint on 3DPW dataset?
